scripts/jwst/build_parent_sample.py

```diff
@@ -267,172 +288,125 @@ def _processing_fn(args):
+                        ],
+                        axis=0,
+                    ).astype(np.float32)
+
+                    # Cutout the center of the image to desired size
+                    s = image.shape
+                    center_x = s[1] // 2
+                    start_x = center_x - image_size // 2
+                    center_y = s[2] // 2
+                    start_y = center_y - image_size // 2
+                    image = image[
+                        :, start_x : start_x + image_size, start_y : start_y + image_size
+                    ]
+                    assert image.shape == (len(filter_list), image_size, image_size), (
+                        "There was an error in reshaping the image to desired size."
+                        "Probably a fiter is missing?"
+                        "Check the available list of filters for the survey",
+                        image.shape,
+                        s,
+                    )
+
+                    # Automatically create _filters by formatting each entry in _filter_list (astropile nomenclature)
+                    filters = [f"jwst_nircam_{filter_name}" for filter_name in filter_list] # Are all of these NIRCam? No MIRI filters?
+                    images.append({
+                        "object_id": id,
+                        "image_band": np.array([f.lower().encode("utf-8") for f in filters], dtype=_utf8_filter_type),
+                        "image_array": image,
+                        "image_psf_fwhm": np.array([0.015 for f in filters]).astype(np.float32), # TODO: Calculate the correct PSF FWHM
+                        "image_scale": np.array([_target_pixel_scale for f in filters]).astype(np.float32),
+                    })
+
+                # Aggregate all images into an astropy table
+                images = Table({k: [d[k] for d in images] for k in images[0].keys()})
+
+                # Join on object_id with the input catalog
+                chunk_catalog = join(chunk, images, keys="object_id", join_type="inner")
+
+                # Making sure we didn't lose anyone
+                assert len(chunk_catalog) == len(images), f"There was an error in the join operation len(chunk_catalog) != len(images): {len(chunk_catalog)} != {len(images)}"
+
+                # Save all columns to disk in HDF5 format
+                with h5py.File(chunk_filename, "w") as hdf5_file:
+                    for key in chunk_catalog.colnames:
+                        hdf5_file.create_dataset(key, data=chunk_catalog[key])
+
+                print(f"... Saved HDF5 chunk: {chunk_filename}")
+
+    print("... Finished processing all groups")
+    return 1
 
-        # Join on object_id with the input catalog
-        catalog = join(catalog, images, keys="object_id", join_type="inner")
 
-        # Making sure we didn't lose anyone
-        assert len(catalog) == len(images), "There was an error in the join operation"
+# Initial survey information
+from info import _SURVEYS_INFO
```

Comment:
Why is this import in the middle of the file?

Reply:
It's not - I was just being a lazy engineer.